synthesis/synthesis.go

```diff
@@ -342,78 +248,67 @@ func FixCds(sqlitePath string, sequence string, codontable codon.Table, problema
 			if !validBias {
 				return sequence, []Change{}, fmt.Errorf("Invalid bias. Expected NA, GC, or AT, got %s", suggestion.Bias)
 			}
-			// First, let's insert the suggestions that we found using our problematicSequenceFuncs
-			_, err := db.Exec(`INSERT INTO suggestedfix(step, start, end, gcbias, quantityfixes, suggestiontype) VALUES (?, ?, ?, ?, ?, ?)`, i, suggestion.Start, suggestion.End, suggestion.Bias, suggestion.QuantityFixes, suggestion.SuggestionType)
-			if err != nil {
-				return sequence, []Change{}, fmt.Errorf("Failed to insert suggestedfix (does start+end point to valid parts of the amino acid sequence?). Got SQL error: %s", err)
+
+			// For each suggestion, get a list of potential changes that could fix the problem.
+			var potentialChanges []Change
+			for positionSelector := suggestion.Start; positionSelector <= suggestion.End; positionSelector++ {
+				lastCodon := historicalMap[positionSelector][len(historicalMap[positionSelector])-1]
+				unavailableCodons := make(map[string]bool)
+				for _, codonSite := range historicalMap[positionSelector] {
+					unavailableCodons[codonSite] = true
+				}
+				// We will take new potential changes from the respective bias map, given the suggestion bias.
+				switch suggestion.Bias {
+				case "NA":
+					for _, potentialCodon := range naBiasMap[lastCodon] {
```

Comment:
A lot of repeat control flow here. Consider making a callable helper function instead of copy and pasting.

Reply:
Fixed!